### DISCUSSION TITLE
[Run issue]: double free or corruption (fasttop) warning

### DISCUSSION THREAD

**Discussion by ghost** [OpenTOPAS/OpenTOPAS#96] [Run-Problem]
### What was the issue?

Hello,

I am running a simulation to calculate the Energy Deposit and Dose to Water for a Cell with GNPs. When I run my code, the terminal output is as follows:

```
TOPAS run sequence complete.
Particle source Demo: Total number of histories: 0
Particle source PhaseSpaceCellFC: Total number of histories: 1229700

Scorer: DoseToWaterAtCell
Results have been written to file: output/CellPhaseSpace/FC/DoseToWaterAtCell.csv
Scored in component: Cell
DoseToWater ( Gy ) : Sum   
0.2608179152011871

Scorer: EnergyDepositAtCell
Results have been written to file: output/CellPhaseSpace/FC/EnergyDepositAtCell.csv
Scored in component: Cell
EnergyDeposit ( MeV ) : Sum   
6.818920612335205

Elapsed times:
Parameter Reading : User=0.000000s Real=0.001720s Sys=0.000000s [Cpu=0.0%]
    Initialization: User=1.220000s Real=2.246706s Sys=1.030000s [Cpu=100.1%]
         Execution: User=90.750000s Real=17.368364s Sys=3.520000s [Cpu=542.8%]
      Finalization: User=0.000000s Real=0.001188s Sys=0.000000s [Cpu=0.0%]
             Total: User=91.97s Real=19.618s Sys=4.55s
double free or corruption (fasttop)
double free or corruption (fasttop)
double free or corruption (fasttop)
double free or corruption (fasttop)
/home/cecilia/shellScripts/topas: line 7: 20382 Segmentation fault      (core dumped) $HOME/Applications/TOPAS/OpenTOPAS-install/bin/topas $1
```

I am not sure what `double free or corruption (fasttop)` means. When I checked my `.csv` scorer output file, it appears normal. Also, based on the elapsed times, it seems the simulation was completed successfully. Should I be concerned about the `double free or corruption (fasttop) warning`, or is it safe to ignore it?

Thank you for your attention.

Best regards,
Cecilia

### TOPAS category

Outcome model

### Version

4.0

### What Operative System (OS) are you running on?

Linux

### Relevant log output

```shell
TOPAS run sequence complete.
Particle source Demo: Total number of histories: 0
Particle source PhaseSpaceCellFC: Total number of histories: 1229700

Scorer: DoseToWaterAtCell
Results have been written to file: output/CellPhaseSpace/FC/DoseToWaterAtCell.csv
Scored in component: Cell
DoseToWater ( Gy ) : Sum   
0.2608179152011871

Scorer: EnergyDepositAtCell
Results have been written to file: output/CellPhaseSpace/FC/EnergyDepositAtCell.csv
Scored in component: Cell
EnergyDeposit ( MeV ) : Sum   
6.818920612335205

Elapsed times:
Parameter Reading : User=0.000000s Real=0.001720s Sys=0.000000s [Cpu=0.0%]
    Initialization: User=1.220000s Real=2.246706s Sys=1.030000s [Cpu=100.1%]
         Execution: User=90.750000s Real=17.368364s Sys=3.520000s [Cpu=542.8%]
      Finalization: User=0.000000s Real=0.001188s Sys=0.000000s [Cpu=0.0%]
             Total: User=91.97s Real=19.618s Sys=4.55s
double free or corruption (fasttop)
double free or corruption (fasttop)
double free or corruption (fasttop)
double free or corruption (fasttop)
/home/cecilia/shellScripts/topas: line 7: 20382 Segmentation fault      (core dumped) $HOME/Applications/TOPAS/OpenTOPAS-install/bin/topas $1
```

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### Attestation

- [X] I testify that I have fully read the bug reporting guideline beforehand.

**Comment by tmasilela** [comment]
Hi Cecilia,

It is not immediately apparent to me what seems to be causing this segmentation fault. If you attach your TOPAS parameter file I can try and take a look when I find some time.

All the best,

Thongchai

**Reply by ghost** [reply]
Hi Thongchai,

Thank you for your response.

Hereby I attached my parameter file, the support file, and the phase space files for your reference. I think the issue might be related to the physics settings I used.

[Cell.txt](https://github.com/user-attachments/files/17811906/Cell.txt)
[TOPASChemistry.txt](https://github.com/user-attachments/files/17811910/TOPASChemistry.txt)
[CellFC_PhaseSpace_compressed.zip](https://github.com/user-attachments/files/17812012/CellFC_PhaseSpace_compressed.zip)

To be honest, I have fixed the issue by changing the physics setting. However, I am still curious about the message `double free or corruption (fasttop)`. Why did it appear? Is this something we can safely ignore?

Thank you for your guidance!

Best,
Cecilia

**Comment by tmasilela** [comment]
Hi Cecilia,

Apologies for the late response. I spent some time looking into this but I'm afraid finding the exact cause of this warning is a bit out of my depth. Although your output appears normal I'm hesitant to say that you can safely ignore this warning.

Since you were able to bypass it by changing the physics list you use I would say to keep using this physics list as long as it is appropriate for your simulations. Good luck!

All the best,
Thongchai
